rocketpy/mathutils/function.py

```diff
@@ -2969,86 +2946,78 @@ def _check_user_input(
             def source_function(_):
                 return temp
 
-            source = source_function
-        return source, inputs, outputs, interpolation, extrapolation
+            return source_function
 
-    @staticmethod
-    def _validate_inputs_outputs(inputs, outputs):
-        """Used to validate the inputs and outputs parameters for creating a
-        Function object. It sets default values if they are not provided.
+        # If source is a callable function
+        return source
+
+    def __validate_inputs(self, inputs):
+        """Used to validate the inputs parameter for creating a Function object.
+        It sets a default value if it is not provided.
 
         Parameters
         ----------
-        inputs : str, list of str, None
+        inputs : list of str, None
             The name(s) of the input variable(s). If None, defaults to "Scalar".
-        outputs :
-            The name of the output variables. If None, defaults to "Scalar".
 
         Returns
         -------
-        tuple
-            A tuple containing the validated inputs and outputs parameters.
-
-        Raises
-        ------
-        ValueError
-            If the output has more than one element.
-        """
-        if inputs is None:
-            inputs = ["Scalar"]
-        if outputs is None:
-            outputs = ["Scalar"]
-        # check output type and dimensions
-        if isinstance(inputs, str):
-            inputs = [inputs]
-        if isinstance(outputs, str):
-            outputs = [outputs]
-        elif len(outputs) > 1:
+        list
+            The validated inputs parameter.
+        """
+        if self.__dom_dim__ == 1:
+            if inputs is None:
+                return ["Scalar"]
+            if isinstance(inputs, str):
+                return [inputs]
+            if isinstance(inputs, list):
+                if len(inputs) == 1 and isinstance(inputs[0], str):
+                    return inputs
```

Comment:
This will not accept `tuples` I think.

Personally, I would not go that strict into checking if the inner type of the Iterable is actually a string. This is the type of check that I do think that is more idiomatic to duck type.